client/dfclient/include/gameplay_state.hpp

```diff
@@ -80,20 +86,28 @@ struct GameplayState
 
 	void updateRemainingText(uint64_t remainingFrames);
 	void updateShadows();
+	void updateHovers(ncine::Vector2f mouseCoords, float radiusSquared);
+
+	template<typename T, typename F>
+	void processMovable(std::map<uint16_t, T>& map, const FlatBuffGenerated::MovableComponent* comp,
+		F createMovableFunc);
+	template<typename T>
+	void deleteUnusedMovables(std::map<uint16_t, T>& map);
```

Comment:
You put declarations of those function templates into the header, but put their definitions into a single cpp file (`gameplay_state.cpp`). Currently, those functions aren't going to work outside the `gameplay_state.cpp` file.

Please put definitions of those templates into the `gameplay_state.hpp` file, for example after the `GameplayState` struct definition.

Reply:
Moved the functions and their dependencies into the header file